wagtail/models/__init__.py

```diff
@@ -2387,6 +2387,9 @@ def permissions_for_user(self, user):
         """
         Return a PagePermissionsTester object defining what actions the user can perform on this page
         """
+        if self.specific_class and not isinstance(self, self.specific_class):
+            # This allows to always call the most specific override for pages
+            return self.specific_deferred.permissions_for_user(user)
```

Comment:
We probably should check whether the method has been overridden before doing this, just to save doing the cast to specific if the specific class doesn't override the method. See how we do it when checking for `preview_modes` in the `is_previewable` method right below this method.

```suggestion
        # Allow specific classes to override this method, but only cast to the
        # specific instance if it's not already specific and if the method has
        # been overridden. This helps improve performance when working with
        # base Page querysets.
        is_overridden = (
            self.specific_class
            and self.specific_class.permissions_for_user
            != type(self).permissions_for_user
        )
        if is_overridden and not isinstance(self, self.specific_class):
            return self.specific_deferred.permissions_for_user(user)
```